### ISSUE TITLE
Feature request for ocrd workspace clone

### ISSUE THREAD

**Issue by VolkerHartmann** [OCR-D/core#97]
Maybe it's a good idea to allow users to clone directory to a given target.
-o --output-dir Clone workspace to given directory (optional)

**Comment by kba** [comment]
You can specify that with `-d` on the workspace:

```
ocrd workspace -d /path-to-target clone -m http://server/mets.xml
```

**Comment by kba** [comment]
Is this in the cookbook?

**Comment by VolkerHartmann** [comment]
WTF? -d is the path to the source for all other commands?
  -d, --directory PATH    Changes the repository folder location.
(In case that your CWD is not the workspace dir. At least I used it that way)
I expected it as a parameter of the clone command!
Maybe a wording problem: 
Usage of the command workspace: Working with workspaces
But in the parameter description: Change the repository...

**Comment by kba** [comment]
The whole "repository" business is because I started this tool with a `git` like behaviour in mind.

Having the directory as a parameter of the `workspace` command allows broad reuse across all subcommands of `workspace`. Also, `git` also has a few top-level flags, such as `-C`.

Give me an example how you want the command to work, we'll adapt it, no reason to be baffled :)

Also your intuition for how working directory and `$PWD` should be related is welcome!

**Comment by VolkerHartmann** [comment]
IMOH ocrd workspace should always work with the current directory holding mets.xml as METS file.
WORKSPACE
Define the workspace.
Options:
-d --workspace-dir Path to workspace (optional) (default .)
-m --mets-file Name of the METS file inside workspaces root directory. (otpional) (default mets.xml)
Examples:
a) ocrd workspace -> cwd containing mets.xml
b) ocrd workspace -d /path/to/workspace -> /path/to/workspace containing mets.xml
c) ocrd workspace -m anotherMets.xml -> cwd containing anotherMets.xml
d) ocrd workspace -d /path/to/workspace -m anotherMets.xml -> /path/to/workspace containing anotherMets.xml
These two parameters define the whole workspace.
The following commands define the actions.
CLONE
Clone workspace to another directory. (name of METS file should be untouched)
All local files (file://...) will be copied to the new directory and METS file will reference to the
new location.
Options: 
-t --target-dir Target directory (will be created) existing directories are not allowed to avoid inconsistencies. (optional, if not defined a tempory directory will be created)
-f --force Overwrites/Remove existing data (optional)
-a --download-all Download all files and change location in METS file (optional)

FIND
Find file(s) inside METS file filtered by (USE, groupID, ID, mimetype). Output is defined by parameter -k
Options: 
  -G, --file-grp TEXT             fileGrp USE
  -m, --mimetype TEXT             Media type to look for
  -g, --group-id TEXT             GROUPID
  -k, --output-field [url|mimetype|groupId|ID|basename|basename_without_extension|local_filename]
                                  Output field
  -a, --download-all              Whether to download found files (if not already local)

PACK
  Pack workspace as ZIP
Options:
-z --zip-file Target file (should not exist) (required)
-f --force Overwrite target file (optional)
-a --download-all (Download and zip also remote files)

UNPACK
  Unpack ZIP as workspace
Options:
-z --zip-file Source file (should not exist) (required)
-t --target-dir Target directory (should not exist) (required)
-f --force Overwrite target directory (optional)

CREATE
  Create a workspace with an empty METS file.
Options:
  -m, --mets-file Name of the METS file. (default: mets.xml)
  -f, --force  Overwrite METS file if it exists

VALIDATE
  Validates workspace
Options:
No further options required.

LIST-GROUP
List the USE attribute of all defined fileGrps.
Options:
No further options required.

ADD
  Add a file to METS in a workspace.
Options:
  -G, --file-grp TEXT  fileGrp USE  [required]
  -i, --file-id TEXT   ID for the file  [required]
  -m, --mimetype TEXT  Media type of the file  [required]
  -u, --url TEXT       url
  -g, --group-id TEXT  GROUPID
  --help               Show this message and exit.

**Comment by VolkerHartmann** [comment]
OCRD-TOOL
Work with ocrd-tool.json
Define the file.
Options:
-d --workspace-dir Path to workspace (optional) (default .)
-f --json-file Name of the OCRD-TOOL file. (otpional) (default ocrd-tool.json)
The following commands define the actions.
LIST-TOOLS
List all tools defined in ocrd-tool.json.
Options:
No further options required.

TOOL
Get information about a specific tool.
Options:
-t, --tool Name of the tool (required)

PARSE-PARAMS
Parse parameters, validate and print the resulting parameters.
Options:
-p --parameters Parameters in JSON format.

VALIDATE
Validate JSON file.
Options:
No further options required.

**Comment by kba** [comment]
The initial issue is resolved by #124 . Feel free to open new issues for CLI refactoring I've missed.
